src/Microsoft.AspNetCore.ResponseCaching/Internal/DistributedResponseCacheStore.cs

```diff
@@ -42,17 +60,51 @@ public DistributedResponseCacheStore(IDistributedCache cache)
             catch { }
         }
 
-        public async Task SetAsync(string key, object entry, TimeSpan validFor)
+        public async Task SetAsync(string key, IResponseCacheEntry entry, TimeSpan validFor)
         {
             try
             {
+                var cachedResponse = entry as CachedResponse;
+                if (cachedResponse != null)
+                {
+                    cachedResponse.BodyKeyPrefix = FastGuid.NewGuid().IdString;
```

Comment:
BodyKeyPrefix is an implementation detail of the store, it shouldn't be on CachedResponse